tests/ode_test.cpp

```diff
@@ -0,0 +1,81 @@
+//
+// test/utilities_test.cpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#include <gnc_ode.hpp>
+
+#include <gtest/gtest.h>
+
+#define PAN_GNC_ODE_TEST_VARS(n) \
+    unsigned int const nt = 4; \
+    unsigned int const ne = 2; \
+    double const t[nt] = { 0.0, 0.1, 0.2, 0.3 }; \
+    double bf[n * ne]; \
+    double _y[nt * ne]; \
+    double *y[nt]; \
+    for (unsigned int i = 0; i < nt; i++) \
+      y[i] = _y + ne * i; \
+    y[0][0] = 1.0; \
+    y[0][1] = 0.0;
+
+static void fsho(double t, double const *y, double *dy) {
+  dy[0] = y[1];
+  dy[1] = -y[0];
+}
+
+TEST(OdeTest, Ode1ShoTest) {
+  PAN_GNC_ODE_TEST_VARS(1)
+  gnc::ode1(t, nt, y, ne, bf, fsho);
+  // Check against MATLAB output
+  ASSERT_FLOAT_EQ(y[1][0],  1.000000000000000);
```

Comment:
I'm amazed this manages to pass, considering rounding errors. Would it be a good idea to add a tolerance for the equality?

Reply:
I mirrored the MATLAB implementations and wrote the tests in doubles so we could be this picky with the tolerances. Gtest has some type of very tight tolerances built in for the `ASSERT_FLOAT_EQ` macro; although, I don't remember of the top of my head what it is. If we start seeing issues down the line we can revisit.